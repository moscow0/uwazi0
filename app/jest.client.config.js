// eslint-disable-next-line import/no-extraneous-dependencies
const { defaults } = require('jest-config');

module.exports = {
  displayName: 'Client',
  testRunner: 'jest-jasmine2',
  testMatch: ['**/react/**/specs/*spec.(j|t)s?(x)'],
  testPathIgnorePatterns: [],
  testEnvironment: 'node',
  setupFilesAfterEnv: ['<rootDir>/setUpJestClient.js'],
  moduleFileExtensions: [...defaults.moduleFileExtensions, 'd.ts'],
  snapshotFormat: {
    escapeString: true,
    printBasicPrototype: true,
  },
  transform: {
    '\\.[jt]sx?$': ['babel-jest', { rootMode: 'upward' }],
  },
  moduleNameMapper: {
    '\\.(css|scss)$': 'identity-obj-proxy',
    '^shared/(.*)': '<rootDir>/shared/$1',
    '^app/(.*)': '<rootDir>/react/$1',
    '^app/UI/(.*)': '<rootDir>/react/UI/$1',
  },
  snapshotSerializers: ['enzyme-to-json/serializer'],
};
