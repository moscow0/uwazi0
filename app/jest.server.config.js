// eslint-disable-next-line import/no-extraneous-dependencies
const { defaults } = require('jest-config');

module.exports = {
  testRunner: 'jest-jasmine2',
  displayName: 'Server',
  testMatch: ['**/api/**/specs/*spec.(j|t)s?(x)', '**/shared/**/specs/*spec.(j|t)s?(x)'],
  testEnvironment: 'node',
  setupFilesAfterEnv: ['<rootDir>/setUpJestServer.js'],
  globalSetup: '<rootDir>/jestServerGlobalSetup.js',
  moduleFileExtensions: [...defaults.moduleFileExtensions, 'd.ts'],
  snapshotFormat: {
    escapeString: true,
    printBasicPrototype: true,
  },
  fakeTimers: {
    legacyFakeTimers: true,
  },
  transform: {
    '\\.[jt]sx?$': ['babel-jest', { rootMode: 'upward' }],
  },
  moduleNameMapper: {
    '^api/(.*)': '<rootDir>/api/$1',
    '^shared/(.*)': '<rootDir>/shared/$1',
  },
};
