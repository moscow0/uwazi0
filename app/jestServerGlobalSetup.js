import { downloadRedis } from './api/utils/downloadRedis.js';

module.exports = async () => {
  downloadRedis();
};
