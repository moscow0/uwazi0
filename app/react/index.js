import React from 'react';
import ReactDOM from 'react-dom';

import * as Sentry from '@sentry/react';

import { BrowserTracing } from '@sentry/tracing';

import { AppContainer } from 'react-hot-loader';

import App from './App';

import './App/sockets';

if (window.SENTRY_APP_DSN) {
  Sentry.init({
    release: window.UWAZI_VERSION,
    environment: window.UWAZI_ENVIRONMENT,
    dsn: window.SENTRY_APP_DSN,
    integrations: [new BrowserTracing()],
    tracesSampleRate: 0.1,
  });
}

const render = Component => {
  ReactDOM.hydrate(
    <AppContainer>
      <Component />
    </AppContainer>,
    document.getElementById('root')
  );
};

render(App);

if (module.hot) {
  module.hot.accept('./App', () => {
    const nextApp = require('./App');
    render(nextApp);
  });
}
