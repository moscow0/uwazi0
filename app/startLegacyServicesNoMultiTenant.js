import { config } from 'api/config';
import { Repeater } from 'api/utils/Repeater';
import { TaskProvider } from 'shared/tasks/tasks';

async function startLegacyServicesNoMultiTenant() {
  if (config.multiTenant || config.clusterMode) {
    return;
  }
  const anHour = 3600000;
  const topicClassificationRepeater = new Repeater(
    () =>
      TaskProvider.runAndWait('TopicClassificationSync', 'TopicClassificationSync', {
        mode: 'onlynew',
        noDryRun: true,
        overwrite: true,
      }),
    anHour
  );
  topicClassificationRepeater.start();
}

export { startLegacyServicesNoMultiTenant };
