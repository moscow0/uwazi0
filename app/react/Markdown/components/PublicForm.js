/* eslint-disable react/jsx-pascal-case */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Immutable from 'immutable';
import { MetadataFormFields, validator, prepareMetadataAndFiles } from 'app/Metadata';
import { LocalForm, actions, Control } from 'react-redux-form';
import { Captcha } from 'app/ReactReduxForms';
import { Translate } from 'app/I18N';
import { publicSubmit } from 'app/Uploads/actions/uploadsActions';
import { bindActionCreators } from 'redux';
import { FormGroup } from 'app/Forms';
import { Icon } from 'UI';
import Loader from 'app/components/Elements/Loader';
import './scss/public-form.scss';
import Dropzone from 'react-dropzone';
import { BrowserView, MobileView } from 'react-device-detect';
import { generateID } from 'shared/IDGenerator';

class PublicForm extends Component {
  static renderTitle(template) {
    const titleProperty = template.get('commonProperties').find(p => p.get('name') === 'title');
    const useGeneratedId = Boolean(titleProperty.get('generatedId'));

    let input = <Control.text id="title" className="form-control" model=".title" />;

    if (useGeneratedId) {
      input = React.cloneElement(input, { defaultValue: generateID(3, 4, 4) });
    }

    return (
      <FormGroup key="title" model=".title">
        <ul className="search__filter">
          <li>
            <label htmlFor="title">
              <Translate context={template.get('_id')}>{titleProperty.get('label')}</Translate>
              <span className="required">*</span>
            </label>
          </li>
          <li className="wide">{input}</li>
        </ul>
      </FormGroup>
    );
  }

  static renderSubmitState() {
    return (
      <div className="public-form submiting">
        <h3>
          <Translate>Submiting</Translate>
        </h3>
        <Loader />
      </div>
    );
  }

  constructor(props) {
    super(props);
    this.removeAttachment = this.removeAttachment.bind(this);
    this.fileDropped = this.fileDropped.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.validators = {
      captcha: { required: val => val && val.text.length },
      ...validator.generate(props.template.toJS()),
    };
    const generatedIdTitle = props.template
      .get('commonProperties')
      .find(p => p.get('name') === 'title' && p.get('generatedId') === true);
    this.state = { submiting: false, files: [], generatedIdTitle };
  }

  fileDropped(files) {
    const uploadedFiles = files;
    this.state.files.forEach(file => uploadedFiles.push(file));
    this.setState({ files: uploadedFiles });
  }

  async removeAttachment(removedFile) {
    await this.setState(prevState => ({
      files: prevState.files.filter(file => file !== removedFile),
    }));
    if (!this.state.files.length) {
      const input = document.querySelector('input[name="publicform.file"]');
      input.value = '';
    }
  }

  attachDispatch(dispatch) {
    this.formDispatch = dispatch;
  }

  resetForm() {
    this.formDispatch(actions.reset('publicform'));
    if (this.state.generatedIdTitle) {
      this.formDispatch(actions.load('publicform', { title: generateID(3, 4, 4) }));
    }
  }

  async handleSubmit(_values) {
    const { submit, remote } = this.props;
    const values = await prepareMetadataAndFiles(
      _values,
      this.state.files,
      this.props.template.toJS()
    );
    try {
      const submitResult = await submit(values, remote);
      this.setState({ submiting: true });
      await submitResult.promise;
      this.resetForm();
      this.setState({ submiting: false, files: [] });
    } catch (e) {
      this.setState({ submiting: false });
    }
    this.refreshCaptcha();
  }

  renderFileField(id, options) {
    const defaults = { className: 'form-control on-mobile', model: `.${id}` };
    const props = Object.assign(defaults, options);
    return (
      <div className="form-group">
        <ul className="search__filter">
          <li className="attachments-list">
            <Translate>{id === 'file' ? 'Document' : 'Attachments'}</Translate>
            <BrowserView>
              <Dropzone
                onDrop={this.fileDropped}
                className="dropzone"
                accept={id === 'file' ? '.pdf' : undefined}
              >
                <label>
                  <div className="text-content">
                    <div id="icon">
                      <Icon icon="cloud-upload-alt" />
                    </div>
                    <div id="upload-text">
                      <Translate>Drop your files here to upload or</Translate>
                    </div>
                    <div id="upload-button">
                      <div id="button">
                        <Translate>Select files on your device</Translate>
                      </div>
                    </div>
                  </div>
                </label>
              </Dropzone>
            </BrowserView>
            <MobileView>
              <Control.file
                id={id}
                {...props}
                onChange={e => this.fileDropped([...e.target.files])}
              />
            </MobileView>
            <div className="preview-list">
              <ul>
                {this.state.files.map(file => (
                  <li key={file.preview}>
                    <div className="preview-title">{file.name}</div>
                    <div>
                      <span onClick={() => this.removeAttachment(file)}>
                        <Icon icon="times" />
                        &nbsp;<Translate>Remove</Translate>
                      </span>
                    </div>
                  </li>
                ))}
              </ul>
            </div>
          </li>
        </ul>
      </div>
    );
  }

  renderCaptcha() {
    return (
      <FormGroup key="captcha" model=".captcha">
        <ul className="search__filter">
          <li>
            <label>
              <Translate>Captcha</Translate>
              <span className="required">*</span>
            </label>
          </li>
          <li className="wide">
            <Captcha
              remote={this.props.remote}
              refresh={refresh => {
                this.refreshCaptcha = refresh;
              }}
              model=".captcha"
            />
          </li>
        </ul>
      </FormGroup>
    );
  }

  render() {
    const { template, thesauris, file, attachments } = this.props;
    const { submiting } = this.state;
    return (
      <LocalForm
        validators={this.validators}
        model="publicform"
        getDispatch={dispatch => this.attachDispatch(dispatch)}
        onSubmit={this.handleSubmit}
      >
        {submiting && PublicForm.renderSubmitState()}
        {!submiting && (
          <div className="public-form">
            {PublicForm.renderTitle(template)}
            <MetadataFormFields
              thesauris={thesauris}
              model="publicform"
              template={template}
              boundChange={(formModel, value) =>
                this.formDispatch(actions.change(formModel, value))
              }
            />
            {file ? this.renderFileField('file', { accept: '.pdf' }) : false}
            {attachments ? this.renderFileField('attachments', { multiple: 'multiple' }) : false}
            {this.renderCaptcha()}
            <button type="submit" className="btn btn-success">
              <Translate>Submit</Translate>
            </button>
          </div>
        )}
      </LocalForm>
    );
  }
}
PublicForm.propTypes = {
  file: PropTypes.bool.isRequired,
  attachments: PropTypes.bool.isRequired,
  remote: PropTypes.bool.isRequired,
  template: PropTypes.instanceOf(Immutable.Map).isRequired,
  thesauris: PropTypes.instanceOf(Immutable.List).isRequired,
  submit: PropTypes.func.isRequired,
};
export const mapStateToProps = (state, props) => ({
  template: state.templates.find(template => template.get('_id') === props.template),
  thesauris: state.thesauris,
  file: props.file !== undefined,
  remote: props.remote !== undefined,
  attachments: props.attachments !== undefined,
});
export function mapDispatchToProps(dispatch) {
  return bindActionCreators({ submit: publicSubmit }, dispatch);
}
export default connect(mapStateToProps, mapDispatchToProps)(PublicForm);
