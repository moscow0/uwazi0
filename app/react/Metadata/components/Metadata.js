/* eslint-disable import/exports-last */
import React from 'react';
import PropTypes from 'prop-types';
import { flattenDeep } from 'lodash';
import { t } from 'app/I18N';
import MarkdownViewer from 'app/Markdown';
import { GroupedGeolocationViewer } from 'app/Metadata/components/GroupedGeolocationViewer';
import GeolocationViewer from './GeolocationViewer';
import { RelationshipLink } from './RelationshipLink';
import ValueList from './ValueList';

const renderRelationshipLinks = (linksProp, compact) => {
  const formattedLinkValues = Array.isArray(linksProp.value) ? linksProp.value : [linksProp.value];
  const hydratedValues = formattedLinkValues.map(linkValue => ({
    value: <RelationshipLink propValue={linkValue} />,
  }));
  const hydratedProp = { ...linksProp, value: hydratedValues };
  return <ValueList compact={compact} property={hydratedProp} />;
};

export const showByType = (prop, compact, templateId) => {
  let result = prop.value;

  switch (prop.type) {
    case null:
      result = t('System', 'No property');
      break;
    case 'markdown':
      result = <MarkdownViewer html markdown={prop.value} />;
      break;
    case 'link':
      result = (
        <a href={prop.value.url} target="_blank" rel="noopener noreferrer">
          {prop.value.label ? prop.value.label : prop.value.url}
        </a>
      );
      break;
    case 'image':
      result = prop.value && (
        <img
          key={prop.value}
          className={`multimedia-img ${prop.style}`}
          src={prop.value}
          alt={prop.label}
        />
      );
      break;
    case 'media': {
      result = prop.value && <MarkdownViewer markdown={`{media}(${prop.value})`} compact />;
      break;
    }
    case 'geolocation':
      result = (
        <GeolocationViewer
          points={prop.value}
          onlyForCards={Boolean(prop.onlyForCards || compact)}
        />
      );
      break;
    case 'select':
      result = prop.parent ? `${prop.parent}: ${prop.value}` : result;
      break;
    case 'geolocation_group':
      result = <GroupedGeolocationViewer members={prop.members} templateId={templateId} />;
      break;
    case 'relationship':
      result = renderRelationshipLinks(prop, compact);
      break;
    default:
      if (prop.value && prop.value.map) {
        const propValue = flattenDeep(
          prop.value.map(_value =>
            _value.parent && Array.isArray(_value.value)
              ? flattenDeep(
                  _value.value.map(v => ({ ...v, value: `${_value.parent}: ${v.value}` }))
                )
              : _value
          )
        );

        // eslint-disable-next-line no-param-reassign
        prop.value = propValue.map(_value => {
          const value = showByType(_value, compact, templateId);
          return value && value.value
            ? value
            : { value, ...(_value.icon !== undefined ? { icon: _value.icon } : {}) };
        });
        result = <ValueList compact={compact} property={prop} />;
      }
      break;
  }

  return result;
};

const computeGroup = (metadata, startIndex) => {
  const members = [];
  let index = startIndex;
  let lastIndexInTemplate = metadata[index].indexInTemplate - 1;

  while (
    index < metadata.length &&
    metadata[index].type === 'geolocation' &&
    metadata[index].indexInTemplate === lastIndexInTemplate + 1
  ) {
    members.push(metadata[index]);
    lastIndexInTemplate = metadata[index].indexInTemplate;
    index += 1;
  }

  return [members, index];
};

const getNewGroupedGeolocationField = members => {
  if (members.length === 1) {
    return {
      type: 'geolocation_group',
      name: 'geolocation_group',
      label: members[0].label,
      translateContext: members[0].translateContext,
      members,
    };
  }

  return {
    type: 'geolocation_group',
    label: 'Combined geolocations',
    members,
  };
};

const groupAdjacentGeolocations = metadata => {
  const groupedMetadata = [];
  let index = 0;

  while (index < metadata.length) {
    if (metadata[index].type !== 'geolocation') {
      groupedMetadata.push(metadata[index]);
      index += 1;
    } else {
      const [members, i] = computeGroup(metadata, index);

      index = i;
      groupedMetadata.push(getNewGroupedGeolocationField(members));
    }
  }

  return groupedMetadata;
};

function filterProps(showSubset) {
  return p => {
    if (showSubset && !showSubset.includes(p.name)) {
      return false;
    }
    if (Array.isArray(p.value)) {
      return p.value.length;
    }
    return p.value || p.type === null || (p.type === 'numeric' && p.value === 0);
  };
}

const flattenInherittedRelationships = metadata =>
  metadata.map(property => {
    if (property.type === 'inherit' && property.inheritedType === 'relationship') {
      // eslint-disable-next-line no-param-reassign
      property.value = property.value.reduce(
        (flattenedValues, v) => flattenedValues.concat(v.value || []),
        []
      );
    }
    return property;
  });

const Metadata = ({
  metadata,
  compact,
  renderLabel,
  showSubset,
  highlight,
  groupGeolocations,
  templateId,
}) => {
  const filteredMetadata = metadata.filter(filterProps(showSubset));
  const flattenedMetadata = flattenInherittedRelationships(filteredMetadata);
  const groupedMetadata = groupGeolocations
    ? groupAdjacentGeolocations(flattenedMetadata)
    : flattenedMetadata;

  return (
    <>
      {groupedMetadata.map((prop, index) => {
        let type = prop.type ? prop.type : 'default';
        type = type === 'image' || type === 'media' ? 'multimedia' : type;
        const highlightClass = highlight.includes(prop.name) ? 'highlight' : '';
        const fullWidthClass = prop.fullWidth ? 'full-width' : '';

        return (
          <dl
            className={`metadata-type-${type} metadata-name-${prop.name} ${fullWidthClass} ${highlightClass}`}
            key={`${prop.name}_${index}`}
          >
            {renderLabel(prop, <dt>{t(prop.translateContext || 'System', prop.label)}</dt>)}
            <dd className={prop.sortedBy ? 'item-current-sort' : ''}>
              {showByType(prop, compact, templateId)}
            </dd>
          </dl>
        );
      })}
    </>
  );
};

Metadata.defaultProps = {
  compact: false,
  showSubset: undefined,
  renderLabel: (_prop, label) => label,
  highlight: [],
  groupGeolocations: false,
};

Metadata.propTypes = {
  metadata: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.string,
      label: PropTypes.string,
      value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
        PropTypes.instanceOf(Object),
        PropTypes.arrayOf(PropTypes.string),
        PropTypes.arrayOf(
          PropTypes.shape({
            value: PropTypes.string,
          })
        ),
      ]),
    })
  ).isRequired,
  templateId: PropTypes.string,
  highlight: PropTypes.arrayOf(PropTypes.string),
  compact: PropTypes.bool,
  renderLabel: PropTypes.func,
  showSubset: PropTypes.arrayOf(PropTypes.string),
  groupGeolocations: PropTypes.bool,
};

export default Metadata;
