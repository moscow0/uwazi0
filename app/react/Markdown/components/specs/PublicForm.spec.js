/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import Immutable from 'immutable';
import { LocalForm } from 'react-redux-form';
import Dropzone from 'react-dropzone';
import { MetadataFormFields } from 'app/Metadata';
import PublicForm from '../PublicForm.js';

describe('PublicForm', () => {
  let props;
  let component;
  let instance;
  let submit;
  let request;

  beforeEach(() => {
    request = Promise.resolve({ promise: Promise.resolve('ok') });
    submit = jasmine.createSpy('submit').and.callFake(async () => request);
  });

  const render = (customProps, generatedId = false) => {
    props = {
      template: Immutable.fromJS({
        _id: '123',
        commonProperties: [
          {
            label: 'Title changed',
            name: 'title',
            generatedId,
          },
        ],
        properties: [
          { type: 'text', name: 'text' },
          { type: 'image', name: 'image' },
        ],
      }),
      thesauris: Immutable.fromJS([]),
      file: false,
      attachments: false,
      submit,
      remote: false,
    };
    const mappedProps = { ...props, ...customProps };
    component = shallow(<PublicForm.WrappedComponent {...mappedProps} />);
    instance = component.instance();
    instance.refreshCaptcha = jasmine.createSpy('refreshCaptcha');
    instance.formDispatch = jasmine.createSpy('formDispatch');
  };

  it('should render a form', () => {
    render();
    expect(component).toMatchSnapshot();
  });

  it('should bind the MetadataFormFields change to this form', () => {
    render();
    const metadataFormFields = component.find(MetadataFormFields);
    metadataFormFields.props().boundChange('publicForm.title', 'New Title');

    expect(instance.formDispatch).toHaveBeenCalledWith(
      expect.objectContaining({ value: 'New Title', model: 'publicForm.title' })
    );
  });

  it('should render a generated ID as title if the option is marked', () => {
    render({}, true);
    const title = component.find('#title').at(0);
    expect(title.props().defaultValue).toEqual(expect.stringMatching(/^[a-zA-Z0-9-]{12}$/));
  });

  it('should load a generated Id as title after submit if the option is marked', async () => {
    render({}, true);
    const title = component.find('#title').at(0);
    expect(title.props().defaultValue).toEqual(expect.stringMatching(/^[a-zA-Z0-9-]{12}$/));

    const formSubmit = component.find(LocalForm).props().onSubmit;
    await formSubmit({ title: 'test' });

    const title1 = component.find('#title').at(0);
    expect(title1.props().defaultValue).toEqual(expect.stringMatching(/^[a-zA-Z0-9-]{12}$/));
    expect(title1).not.toEqual(title);
  });

  it('should enable remote captcha', () => {
    render({ remote: true });
    expect(component).toMatchSnapshot();
  });

  it('should render a form with file and attachments', () => {
    render({ file: true, attachments: true });
    expect(component).toMatchSnapshot();
  });

  it('should submit the values wrapped', async () => {
    render();

    const formSubmit = component.find(LocalForm).props().onSubmit;
    await formSubmit({ title: 'test', metadata: { color: 'red', size: 42, date: 13442423 } });

    expect(props.submit).toHaveBeenCalledWith(
      {
        file: undefined,
        metadata: { color: [{ value: 'red' }], date: [{ value: 13442423 }], size: [{ value: 42 }] },
        template: '123',
        title: 'test',
        attachments: [],
      },
      false
    );
  });

  it('should refresh the captcha and clear the form after submit', async () => {
    render();
    const formSubmit = component.find(LocalForm).props().onSubmit;
    await formSubmit({ title: 'test' });

    expect(instance.formDispatch).toHaveBeenCalledWith({
      model: 'publicform',
      type: 'rrf/reset',
    });
    expect(instance.refreshCaptcha).toHaveBeenCalled();
  });

  it('should refresh captcha and NOT clear the form on submission error', async () => {
    request = new Promise(resolve => {
      resolve({ promise: Promise.reject() });
    });
    submit = jasmine.createSpy('submit').and.returnValue(request);
    render();
    const formSubmit = component.find(LocalForm).props().onSubmit;
    await formSubmit({ title: 'test' });

    request.then(uploadCompletePromise => {
      uploadCompletePromise.promise
        .then(() => fail('should throw error'))
        .catch(() => {
          expect(instance.formDispatch).not.toHaveBeenCalledWith();
          expect(instance.refreshCaptcha).toHaveBeenCalled();
        });
    });
  });

  it('should keep attachments after a submission error', async done => {
    request = new Promise(resolve => {
      resolve({ promise: Promise.reject() });
    });
    render({ attachments: true });
    const formSubmit = component.find(LocalForm).props().onSubmit;
    const newFile = new File([Buffer.from('image').toString('base64')], 'image.jpg', {
      type: 'image/jpg',
    });
    const attachments = component.find('.preview-title');
    expect(attachments.length).toEqual(0);

    component.find(Dropzone).simulate('drop', [newFile]);
    await formSubmit({ title: 'test' });
    request.then(uploadCompletePromise => {
      uploadCompletePromise.promise
        .then(() => fail('should throw error'))
        .catch(() => {
          const actualAttachments = component.find('.preview-title');
          expect(actualAttachments.length).toBe(1);
          expect(actualAttachments.get(0).props.children).toEqual('image.jpg');
          expect(instance.formDispatch).not.toHaveBeenCalledWith();
          expect(instance.refreshCaptcha).toHaveBeenCalled();
          done();
        });
    });
  });
});
