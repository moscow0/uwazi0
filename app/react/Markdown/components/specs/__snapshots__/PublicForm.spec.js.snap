// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicForm should enable remote captcha 1`] = `
<LocalForm
  getDispatch={[Function]}
  initialState={Object {}}
  model="publicform"
  onSubmit={[Function]}
  validators={
    Object {
      "captcha": Object {
        "required": [Function],
      },
      "title": Object {
        "required": [Function],
      },
    }
  }
>
  <div
    className="public-form"
  >
    <Connect(FormGroup)
      key="title"
      model=".title"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label
            htmlFor="title"
          >
            <Connect(Translate)
              context="123"
            >
              Title changed
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <DefaultConnectedControlText
            className="form-control"
            id="title"
            model=".title"
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <Connect(MetadataFormFields)
      boundChange={[Function]}
      model="publicform"
      template={
        Immutable.Map {
          "_id": "123",
          "commonProperties": Immutable.List [
            Immutable.Map {
              "label": "Title changed",
              "name": "title",
              "generatedId": false,
            },
          ],
          "properties": Immutable.List [
            Immutable.Map {
              "type": "text",
              "name": "text",
            },
            Immutable.Map {
              "type": "image",
              "name": "image",
            },
          ],
        }
      }
      thesauris={Immutable.List []}
    />
    <Connect(FormGroup)
      key="captcha"
      model=".captcha"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label>
            <Connect(Translate)>
              Captcha
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <Captcha
            model=".captcha"
            refresh={[Function]}
            remote={true}
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <button
      className="btn btn-success"
      type="submit"
    >
      <Connect(Translate)>
        Submit
      </Connect(Translate)>
    </button>
  </div>
</LocalForm>
`;

exports[`PublicForm should render a form 1`] = `
<LocalForm
  getDispatch={[Function]}
  initialState={Object {}}
  model="publicform"
  onSubmit={[Function]}
  validators={
    Object {
      "captcha": Object {
        "required": [Function],
      },
      "title": Object {
        "required": [Function],
      },
    }
  }
>
  <div
    className="public-form"
  >
    <Connect(FormGroup)
      key="title"
      model=".title"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label
            htmlFor="title"
          >
            <Connect(Translate)
              context="123"
            >
              Title changed
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <DefaultConnectedControlText
            className="form-control"
            id="title"
            model=".title"
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <Connect(MetadataFormFields)
      boundChange={[Function]}
      model="publicform"
      template={
        Immutable.Map {
          "_id": "123",
          "commonProperties": Immutable.List [
            Immutable.Map {
              "label": "Title changed",
              "name": "title",
              "generatedId": false,
            },
          ],
          "properties": Immutable.List [
            Immutable.Map {
              "type": "text",
              "name": "text",
            },
            Immutable.Map {
              "type": "image",
              "name": "image",
            },
          ],
        }
      }
      thesauris={Immutable.List []}
    />
    <Connect(FormGroup)
      key="captcha"
      model=".captcha"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label>
            <Connect(Translate)>
              Captcha
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <Captcha
            model=".captcha"
            refresh={[Function]}
            remote={false}
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <button
      className="btn btn-success"
      type="submit"
    >
      <Connect(Translate)>
        Submit
      </Connect(Translate)>
    </button>
  </div>
</LocalForm>
`;

exports[`PublicForm should render a form with file and attachments 1`] = `
<LocalForm
  getDispatch={[Function]}
  initialState={Object {}}
  model="publicform"
  onSubmit={[Function]}
  validators={
    Object {
      "captcha": Object {
        "required": [Function],
      },
      "title": Object {
        "required": [Function],
      },
    }
  }
>
  <div
    className="public-form"
  >
    <Connect(FormGroup)
      key="title"
      model=".title"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label
            htmlFor="title"
          >
            <Connect(Translate)
              context="123"
            >
              Title changed
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <DefaultConnectedControlText
            className="form-control"
            id="title"
            model=".title"
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <Connect(MetadataFormFields)
      boundChange={[Function]}
      model="publicform"
      template={
        Immutable.Map {
          "_id": "123",
          "commonProperties": Immutable.List [
            Immutable.Map {
              "label": "Title changed",
              "name": "title",
              "generatedId": false,
            },
          ],
          "properties": Immutable.List [
            Immutable.Map {
              "type": "text",
              "name": "text",
            },
            Immutable.Map {
              "type": "image",
              "name": "image",
            },
          ],
        }
      }
      thesauris={Immutable.List []}
    />
    <div
      className="form-group"
    >
      <ul
        className="search__filter"
      >
        <li
          className="attachments-list"
        >
          <Connect(Translate)>
            Document
          </Connect(Translate)>
          <BrowserView>
            <t
              accept=".pdf"
              className="dropzone"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={true}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              <label>
                <div
                  className="text-content"
                >
                  <div
                    id="icon"
                  >
                    <Connect(Icon)
                      icon="cloud-upload-alt"
                    />
                  </div>
                  <div
                    id="upload-text"
                  >
                    <Connect(Translate)>
                      Drop your files here to upload or
                    </Connect(Translate)>
                  </div>
                  <div
                    id="upload-button"
                  >
                    <div
                      id="button"
                    >
                      <Connect(Translate)>
                        Select files on your device
                      </Connect(Translate)>
                    </div>
                  </div>
                </div>
              </label>
            </t>
          </BrowserView>
          <MobileView>
            <DefaultConnectedControlFile
              accept=".pdf"
              className="form-control on-mobile"
              id="file"
              model=".file"
              onChange={[Function]}
            />
          </MobileView>
          <div
            className="preview-list"
          >
            <ul />
          </div>
        </li>
      </ul>
    </div>
    <div
      className="form-group"
    >
      <ul
        className="search__filter"
      >
        <li
          className="attachments-list"
        >
          <Connect(Translate)>
            Attachments
          </Connect(Translate)>
          <BrowserView>
            <t
              className="dropzone"
              disableClick={false}
              disablePreview={false}
              disabled={false}
              maxSize={Infinity}
              minSize={0}
              multiple={true}
              onDrop={[Function]}
              preventDropOnDocument={true}
            >
              <label>
                <div
                  className="text-content"
                >
                  <div
                    id="icon"
                  >
                    <Connect(Icon)
                      icon="cloud-upload-alt"
                    />
                  </div>
                  <div
                    id="upload-text"
                  >
                    <Connect(Translate)>
                      Drop your files here to upload or
                    </Connect(Translate)>
                  </div>
                  <div
                    id="upload-button"
                  >
                    <div
                      id="button"
                    >
                      <Connect(Translate)>
                        Select files on your device
                      </Connect(Translate)>
                    </div>
                  </div>
                </div>
              </label>
            </t>
          </BrowserView>
          <MobileView>
            <DefaultConnectedControlFile
              className="form-control on-mobile"
              id="attachments"
              model=".attachments"
              multiple="multiple"
              onChange={[Function]}
            />
          </MobileView>
          <div
            className="preview-list"
          >
            <ul />
          </div>
        </li>
      </ul>
    </div>
    <Connect(FormGroup)
      key="captcha"
      model=".captcha"
    >
      <ul
        className="search__filter"
      >
        <li>
          <label>
            <Connect(Translate)>
              Captcha
            </Connect(Translate)>
            <span
              className="required"
            >
              *
            </span>
          </label>
        </li>
        <li
          className="wide"
        >
          <Captcha
            model=".captcha"
            refresh={[Function]}
            remote={false}
          />
        </li>
      </ul>
    </Connect(FormGroup)>
    <button
      className="btn btn-success"
      type="submit"
    >
      <Connect(Translate)>
        Submit
      </Connect(Translate)>
    </button>
  </div>
</LocalForm>
`;
